### PR TITLE
add racc modifier for shigeto bow

### PR DESCRIPTION
**Pull request by lapislosh**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Adds RACC modifier for Shigeto bow for SAM only, which required a new latent effect for main job (previously only subjob-based latents existed)

https://ffxiclopedia.fandom.com/wiki/Shigeto_Bow

## Steps to test these changes

!additem 18142
!additem 18143

Change job to rng and then same and compare racc with/without bow. RNG RACC should remained unchanged while SAM RACC should increase.
```
Job	Weapon		RACC / RATK
SAM	Shortbow	50 / 71
SAM	Shigeto		57 / 86
SAM	Shigeto+1	58 / 87
RNG	Shortbow	103 / 67
RNG	Shigeto		103 / 82
RNG	Shigeto+1	103 / 83
```

### CODE REVIEW

**Review by TeoTwawki — APPROVED** [review]

**Review by EpicTaru — APPROVED** [review]

File: src/map/latent_effect.h [hunk 1]
```diff
@@ -91,6 +91,7 @@ enum class LATENT : uint16
     VS_ECOSYSTEM          = 59, // Vs. Specific Ecosystem ID (e.g. Vs. Plantoid: Accuracy+3)
     VS_FAMILY             = 60, // Vs. Specific Family ID (e.g. Vs. Korrigan: Accuracy+3)
     VS_SUPERFAMILY        = 61, // Vs. Specific SuperFamily ID (e.g. Vs. Mandragora: Accuracy+3)
+    MAINJOB               = 62, // mainjob - PARAM: JOBTYPE
 };
 
 #define MAX_LATENTEFFECTID 61
```
Comment (EpicTaru) [inline src/map/latent_effect.h]: Doesn't this value also need to be updated?

Comment (TeoTwawki) [reply src/map/latent_effect.h]: good catch, it does. And looks like it was missed previously or overwritten in prior pr also\*, it should be 63 as of this pr.

\* probably 2 pr's came in at same time and conflicted on this spot

Comment (lapislosh) [reply src/map/latent_effect.h]: yes and no, I actually started it before I read this comment!
the value doesn't seem to be used anything but should be updated nonetheless

Comment (EpicTaru) [reply src/map/latent_effect.h]: For clarity's sake, I'm referring to Line 97, doesn't the MAX_LATENTEFFECTID value need updated?

Comment (lapislosh) [reply src/map/latent_effect.h]: ah you are correct, it starts at 0 and was missed... I'll fix it to 63 then

Comment (lapislosh) [reply src/map/latent_effect.h]: orrrrrr it technically says "max latent ID" not "count" maybe it is supposed to be 62? maybe I should just delete it

Comment (TeoTwawki) [reply src/map/latent_effect.h]: the way we use max IDs requires them to be 1+highest used ID.

(this is one reason I hate what we name that and also hate that it exists, and am happy anytime one gets refactored out.. sometimes the alternative to it has other negative impacts tho)

Comment (TeoTwawki) [reply src/map/latent_effect.h]: If you removed that max id line, the server would implode in odd places. its necessary.

Comment (lapislosh) [reply src/map/latent_effect.h]: I guess we'll leave it at 63 and hope for the best

Comment (TeoTwawki) [reply src/map/latent_effect.h]: 63 is correct, so that is the best outcome.

**Review by EpicTaru — APPROVED** [review]

**Comment by lapislosh** [comment]
pushed one last time to reduce down to 1 commit

**Comment by TeoTwawki** [comment]
> pushed one last time to reduce down to 1 commit

bless you kind sir

**Review by TeoTwawki — APPROVED** [review]

**Pull request merged**